packages/zoe/src/contractFacet/zcfZygote.js

```diff
@@ -152,31 +153,64 @@ export const makeZCFZygote = (
           zcfSeat = makeEmptySeatKit().zcfSeat;
         }
         const totalToMint = Object.values(gains).reduce(add, empty);
-        zcfSeat.incrementBy(gains);
+        assert(
+          !zcfSeat.hasExited(),
+          `zcfSeat must be active to mint gains for the zcfSeat`,
+        );
+        const allocationPlusGains = addToAllocation(
+          zcfSeat.getCurrentAllocation(),
+          gains,
+        );
+
+        // Increment the stagedAllocation if it exists so that the
+        // stagedAllocation is kept up to the currentAllocation
+        if (zcfSeat.hasStagedAllocation()) {
+          zcfSeat.incrementBy(gains);
+        }
+
         // verifies offer safety
-        assert(zcfSeat.isOfferSafe(zcfSeat.getStagedAllocation()));
-        // No effects above. The following two steps
-        // *should* be committed atomically, but it is not a
-        // disaster if they are not. If we minted only, no one would
-        // ever get those invisibly-minted assets.
+        assert(
+          zcfSeat.isOfferSafe(allocationPlusGains),
+          `The allocation after minting gains ${allocationPlusGains} for the zcfSeat was not offer safe`,
+        );
```

Comment:
Same question applies both before and after: Aside from a bug in zoe we don't know about, is there any way for this to fail? Since the previous allocation was offer safe, adding gains to it should not be able to make it not be offer safe, yes?

If so, I am not suggesting removing the test. But perhaps rewording it to indicate an internal error in Zoe, rather than a mistake in the contract.

Reply:
There's no way for this to fail apart from a bug in Zoe, but I wanted to leave it in so it's easy to say that reallocations pass through an offer safety check. I'll add a comment about the unfeasibility of this failing unless something very wrong happens.

Reply:
Comment added